src/commands/dev/lib/dev-server.ts

```diff
@@ -184,26 +185,84 @@ export default class DevServer {
     );
   }
 
-  async send404(res: http.ServerResponse): Promise<void> {
+  async send404(
+    req: http.IncomingMessage,
+    res: http.ServerResponse,
+    nowRequestId: string
+  ): Promise<void> {
     return this.sendError(
+      req,
       res,
+      nowRequestId,
       'FILE_NOT_FOUND',
       'The page could not be found',
       404
     );
   }
 
   async sendError(
+    req: http.IncomingMessage,
     res: http.ServerResponse,
+    nowRequestId: string,
     code: string,
     message: string,
     statusCode: number = 500
   ): Promise<void> {
-    // TODO: render an HTML page similar to Now's router
     res.statusCode = statusCode;
+    this.writeResponseHeaders(res, nowRequestId);
+    // TODO: render an HTML page similar to Now's router
     res.end(`${statusCode}: ${message}\nCode: ${code}\n`);
   }
 
+  getRequestIp(req: http.IncomingMessage): string {
+    // TODO: respect the `x-forwarded-for` headers
+    return req.connection.remoteAddress || '127.0.0.1';
+  }
+
+  /**
+   * Sets the response `headers` including the Now headers to `res`.
+   */
+  writeResponseHeaders(
```

Comment:
Since we use `res.setHeader` other than `res.writeHead` in this function, maybe it's better name as `setResponseHeaders`?

Reply:
Sure thing